inbm/dispatcher-agent/tests/unit/sota/test_sota.py

```diff
@@ -270,3 +271,100 @@ def test_is_ota_no_update_available_return_false(self) -> None:
 Processing triggers for systemd (245.4-4ubuntu3.23) ...
         """
         self.assertFalse(TestSota.sota_instance._is_ota_no_update_available(cmd_list))
+
+    @patch('dispatcher.sota.sota.get_os_version', return_value='2.0.20240802.0213')
+    @patch('inbm_common_lib.shell_runner.PseudoShellRunner.run', return_value=("TiberOS", "", 0))
+    def test_save_granular_in_tiberos_with_success_log(self, mock_run, mock_get_os_version) -> None:
+        TestSota.sota_instance.factory = SotaOsFactory(
+            TestSota.mock_disp_broker,
+            None, []).get_os('TiberOS')
+
+        TestSota.sota_instance._update_logger.detail_status = OTA_SUCCESS
+        TestSota.sota_instance.save_granular_log(check_package=False)
+
+        with open(GRANULAR_LOG_FILE, 'r') as f:
+            granular_log = json.load(f)
+
+        expected_content = {
+            "UpdateLog": [
+                {
+                    "StatusDetail.Status": OTA_SUCCESS,
+                    "OS Version": '2.0.20240802.0213'
+                }
+            ]
+        }
+
+        self.assertEqual(granular_log, expected_content)
+
+    @patch('dispatcher.sota.sota.get_os_version', return_value='2.0.20240802.0213')
+    @patch('inbm_common_lib.shell_runner.PseudoShellRunner.run', return_value=("TiberOS", "", 0))
+    def test_save_granular_in_tiberos_with_pending_log(self, mock_run, mock_get_os_version) -> None:
+        TestSota.sota_instance.factory = SotaOsFactory(
+            TestSota.mock_disp_broker,
+            None, []).get_os('TiberOS')
+
+        TestSota.sota_instance._update_logger.detail_status = OTA_PENDING
+        TestSota.sota_instance.save_granular_log(check_package=False)
+
+        with open(GRANULAR_LOG_FILE, 'r') as f:
```

Comment:
interacting with fs in unit test

Reply:
Updated. Use mock_open  for the file operation and verify the parameter passed in json.dump.